gluon/tools.py

```diff
@@ -1393,16 +1400,16 @@ def api_auth():
             username = request.vars[self.user_param]
             password = request.vars[self.pass_param]
             valid_user = self.auth.login_bare(username, password)
+        else:
+            valid_user = self.auth.user
             if valid_user:
                 payload = self.serialize_auth_session(current.session.auth)
                 self.alter_payload(payload)
                 ret = {'token':self.generate_token(payload)}
             else:
                 raise HTTP(
-                    401, u'Not Authorized', 
+                    401, u'Not Authorized',
                     **{'WWW-Authenticate': u'JWT realm="%s"' % self.realm})
-        else:
```

Comment:
It seems I've messed the PR. There is some indentation mess.
The idea was to keep only the 'Not Authorized' redirection…
I'll have a second take to this PR… after a good night.